system-tests/e2e-suite/src/test/scala/com/hotels/styx/proxy/https/EmptyTlsProtocolsListSpec.scala

```diff
@@ -0,0 +1,90 @@
+/**
+  * Copyright (C) 2013-2017 Expedia Inc.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License");
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  * http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+  */
+package com.hotels.styx.proxy.https
+
+import com.github.tomakehurst.wiremock.client.WireMock._
+import com.hotels.styx.api.client.UrlConnectionHttpClient
+import com.hotels.styx.api.{HttpClient, HttpRequest}
+import com.hotels.styx.infrastructure.HttpResponseImplicits
+import com.hotels.styx.support.ResourcePaths.fixturesHome
+import com.hotels.styx.support.backends.FakeHttpServer
+import com.hotels.styx.support.configuration._
+import com.hotels.styx.{SSLSetup, StyxProxySpec}
+import io.netty.handler.codec.http.HttpHeaders.Names._
+import io.netty.handler.codec.http.HttpMethod._
+import io.netty.handler.codec.http.HttpResponseStatus.OK
+import org.scalatest.{FunSpec, ShouldMatchers}
+
+class EmptyTlsProtocolsListSpec extends FunSpec
+  with StyxProxySpec
+  with HttpResponseImplicits
+  with ShouldMatchers
+  with SSLSetup {
+
+  val crtFile = fixturesHome(this.getClass, "/ssl/testCredentials.crt").toString
+  val keyFile = fixturesHome(this.getClass, "/ssl/testCredentials.key").toString
+  val clientV12 = newClient("TLSv1.2")
+  val clientV11 = newClient("TLSv1.1")
+
+  override val styxConfig = StyxConfig(
+    ProxyConfig(
+      Connectors(
+        HttpConnectorConfig(),
+        HttpsConnectorConfig(
+          certificateFile = crtFile,
+          certificateKeyFile = keyFile,
+          protocols = List()
+        ))
+    )
+  )
+
+  val recordingBackend = FakeHttpServer.HttpsStartupConfig(needClientAuth = false)
+    .start()
+
+  override protected def beforeAll() = {
+    super.beforeAll()
+    styxServer.setBackends("/secure" -> HttpsBackend(
+      "https-app", Origins(recordingBackend), TlsSettings()
+    ))
+  }
+
+  override protected def afterAll() = {
+    recordingBackend.stop()
+    super.afterAll()
+  }
+
+  describe("Empty TLS protocols list") {
+    recordingBackend.stub(urlPathEqualTo("/secure"), aResponse.withStatus(200))
+
+    it("Empty TLS protocols list activates all supported protocols") {
+      val req = new HttpRequest.Builder(GET, styxServer.secureRouterURL("/secure"))
```

Comment:
Why not simply import `HttpRequest.Builder.get` instead of using this verbose alternative?

Reply:
Addressed all comments. Please have another look.